File: python/src/problems/bubble_sort.py
def log_swaps(a) -> str:
    swaps = 0

    for i in range(0, len(a)):
        for j in range(0, len(a) - i - 1):

            if a[j] > a[j + 1]:
                temp = a[j + 1]
                a[j + 1] = a[j]
                a[j] = temp
                swaps += 1

    return f"""
      Array is sorted in {swaps} swaps.
      First Element: {a[0]}
      Last Element: {a[len(a) - 1]}
    """

File: javascript/src/bubbleSort.js


function bubbleSort(a) {

  let swaps = 0;
  for(let i = 0; i < a.length; i++) {
    for( let j = 0; j < a.length - i - 1; j++) {
      if(a[j] > a[j + 1]) {
        const temp = a[j];
        a[j] = a[j + 1];
        a[j + 1] = temp;
        swaps++;
      }
    }
  }

  const result = `Array is sorted in ${swaps} swaps.
  First Element: ${a[0]}
  Last Element: ${a[a.length - 1]}
  `
  return result
}

module.exports = bubbleSort